### ISSUE TITLE
[GPU Error Bug] SELECT <column> FROM JOIN tables ON (CAST(<decimal> AS MONEY)) Brings GPU Errors

### ISSUE THREAD

**Issue by qwebug** [heterodb/pg-strom#626]
### Describe:

Using SELECT table2\<column> FROM table1 RIGHT OUTER JOIN table2 ON (CAST(\<decimal> AS MONEY) IN CAST(\<decimal> AS MONEY)) will report an GPU error, when pg\_strom.enabled is turned on.

However it is able to output result, when pg\_strom.enabled is turned off.

#### SQL < pg\_strom.enabled ON >:

```sql
CREATE TABLE t1(c0 INT);
CREATE TABLE t2(LIKE t1);
INSERT INTO t2(c0) VALUES(1);
INSERT INTO t1(c0) VALUES(1);
CREATE SCHEMA extensions;
CREATE EXTENSION pg_strom WITH SCHEMA extensions;
set pg_strom.enabled=on;
SELECT t2.c0 FROM t1 RIGHT OUTER JOIN t2 ON ((CAST(0.1 AS MONEY)) IN (CAST(0.1 AS MONEY)));
```

Result:

```bash
ERROR:  gpu_service.c:1794  failed on cuEventSynchronize: CUDA_ERROR_ILLEGAL_ADDRESS
HINT:  device at GPU-0, function at gpuservHandleGpuTaskExec
```

#### SQL < pg\_strom.enabled OFF >:

```sql
set pg_strom.enabled=off;
SELECT t2.c0 FROM t1 RIGHT OUTER JOIN t2 ON ((CAST(0.1 AS MONEY)) IN (CAST(0.1 AS MONEY)));
```

Result:

```bash
 c0 
----
  1
(1 row)
```

Futhermore, if change RIGHT OUTER JOIN to LEFT OUTER JOIN or INNER JOIN, errors also occur while pg\_strom.enabled is turned on.

### Environment:

Pg-strom Version: commit 4ac6d95339494f901b93acb384186415dfe63c1b

PostgreSQL Version: 15.3 on x86\_64-pc-linux-gnu, compiled by gcc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-18), 64-bit

CUDA Version: 12.0.1

NVIDIA Driver Version: 535.86.05

**Comment by kaigai** [comment]
Thanks for this report. This problem was fixed at `78002d17a1a1fd7dbdeaecd5b5b5e181dc17933c`

If join-clause contains no inner-/outer- var references, it works as a cross-join or an empty-join. Regardless of the results, they are not welcome for xPU-Join workloads.

**Comment by qwebug** [comment]
This problem is fixed, after my verification.
Thanks for your work.
